README.md

```diff
@@ -43,12 +42,20 @@ On Windows, you can do this in Docker Desktop>Settings>Build engine.
 On Mac, you can do this in Docker Desktop>Preferences>Docker Engine. Edit the displayed JSON so that `"buildkit": false`, then restart Docker Desktop.
 ## Test
 Run `script/test`, or follow these steps:
-1. Run `script/bootstrap` script
-2. Install dev requirements using `pip install -r requirements-dev.txt`
+1. Run `npm install`
+2. Run `script/bootstrap` script
 3. Start everything with `docker compose up`.
 4. Once all containers are initialised and healthy, run the tests by typing `script/cucumber`
 
-To run a specific feature, add the path to the specific .feature file to the `script/cucumber` script. For example, to run the selfdialog feature, the following command has to be used: `./node_modules/.bin/cucumber-js features/selfdialog.feature`
+To run a specific feature, add the path to the specific .feature file to the `script/cucumber` script. 
+For example, to run the selfdialog feature, the following command has to be used: `./node_modules/.bin/cucumber-js features/selfdialog.feature`
+
+### Updating the testing routine 
+
+The testing procedure is build using [cucumber](https://cucumber.io/docs). 
+By running the tests, the features implemented in the .feature files contained in feature folder are executed, using the implementation contained in `features/agenda_steps.js`.
+Each scenario in the feature represents all the steps of one dialog, and the testing executes all of them and verifies that the result is the expected one.
+In case of modifications to the dialogs, the testing steps have to be updated accordingly, by modifying the steps in the .feature file and their implementation in the agenda_steps.js file.
```

Comment:
What I did not know when I first looked at the testing code was how to handle custom actions that just set slots. From what I understand, we do not currently check if slots have the right value. Maybe that would be helpful to add?

Reply:
This is actually a very good point. Currently the slots content is checked indirectly by verifying the content of response that makes use of it, but there might be cases where this is not possible. I will open an issue for this.

Reply:
Of course, we need to investigate if this is possible in the first place :)